rpcd/playbooks/roles/rpc_pre_upgrade/tasks/main.yml

```diff
@@ -25,6 +25,7 @@
 
 # Gather swift-recon md5 output
 - include: gather_swift_stats.yml
+  when: groups['swift_proxy']|default([])|length > 0
```

Comment:
Well, pre-upgrades playbook runs on localhost (which is terrible IMO but that's another story), where post-upgrades run on galera_all:rabbitmq:utility:swift_proxy:elasticsearch_container (which is less terrible but let's go on).
In that sense, the post_upgrades "inventory_hostname in groups['whatever']" will never work on pre-upgrades.

Let's continue with what's needed. Swift, log rotation isn't forced to be deployed, and the deployer could even have removed the env.d, so it's good to ensure that the group exists and have nodes in it before doing anything.

Two methods are good for that.
```
when:
  - groups['swift_proxy'] is defined
  - groups['swift_proxy']|length > 0

```
or the shorter version:
`when: groups['swift_proxy']|default([])|length > 0`